nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java

```diff
@@ -579,6 +584,44 @@ public void testGCEnabled() {
             "Cannot expire snapshots: GC is disabled (deleting files may corrupt other tables)");
   }
 
+  @Test
+  public void testTableMetadataFilesCleanupDisable() {
+    Table icebergTable = catalog.loadTable(TABLE_IDENTIFIER);
+
+    Assertions.assertThat(
+            icebergTable.properties().get(TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED))
+        .isNotNull()
+        .isEqualTo("false");
+
+    icebergTable
+        .updateProperties()
+        .set(TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED, "true")
```

Comment:
Is this expected to be overridden back to `false` by the Nessie Catalog? If so, shouldn't we assert that?